scarlet2/observation.py

```diff
@@ -36,6 +36,10 @@ def __init__(self, data, weights, psf=None, wcs=None, channels=None, renderer=No
     def C(self):
         return self.frame.C
 
+    def set_weights(self, weights):
```

Comment:
I don't think this method is needed. Simply specify the weights at init time. They are not supposed to change afterwards anyway (this was just a little sloppy in the old docs)

Reply:
OK, I see why this was done in the docs. We simply didn't have the background rms for these observations without running `sep`. I'd still prefer to remove this method and instead rewrite the method `makeCatalog` from the notebook to accept the images (aka `obs.data`) instead.